### ISSUE TITLE
Headstart: Consider launching the Headstart-on-Demand feature.

### ISSUE THREAD

**Issue by kwight** [Automattic/wp-calypso#52760]
"Headstart on Demand" is a feature that allows users themselves to reset their site to look like the theme demo. It's currently available on the `wpcalypso` environment at `/settings/theme-setup/<site>`.

<img width="700" alt="Screen Shot 2021-05-04 at 2 51 24 PM" src="https://user-images.githubusercontent.com/349751/117877975-168be600-b25a-11eb-8038-8181c1d2fc25.png">

**Comment by kwight** [comment]
The feature hasn't been maintained for a couple of years (and I think pre-dates Atomic sites) – extra care will be needed in testing with our various plans and environments.

**Comment by michaeldcain** [comment]
With the updates the Design Selection pod made to the Headstart annotations for our most popular themes, I figured I'd just see what would happen on an existing site if I used the feature on wpcalypso (Settings > Theme Setup). 

**tl;dr:** it works quite well: Pd4NFJ-tJ-p2

**Before:**
<img width="1755" alt="Screen Shot 2021-07-16 at 2 11 22 PM" src="https://user-images.githubusercontent.com/942359/125991925-175fb010-df49-4616-8f05-437be083ff9a.png">
<img width="1755" alt="Screen Shot 2021-07-16 at 2 11 11 PM" src="https://user-images.githubusercontent.com/942359/125991931-96d6d0d5-5dca-4f45-ad71-f513c7085cd8.png">
<img width="1755" alt="Screen Shot 2021-07-16 at 2 11 03 PM" src="https://user-images.githubusercontent.com/942359/125991933-074f7489-55d1-44c7-bcb9-55a454dac23a.png">
<img width="1755" alt="Screen Shot 2021-07-16 at 2 10 57 PM" src="https://user-images.githubusercontent.com/942359/125991937-ba68384c-f7cc-4633-8e9d-ad9658d4bd9d.png">

**During:**
<img width="1755" alt="Screen Shot 2021-07-16 at 2 12 09 PM" src="https://user-images.githubusercontent.com/942359/125991983-6d53f836-f18f-4b48-81eb-7ca0ca9ece96.png">
<img width="1755" alt="Screen Shot 2021-07-16 at 2 12 12 PM" src="https://user-images.githubusercontent.com/942359/125991980-e65bb79a-5090-49f4-be17-ae0c7b140b88.png">
<img width="1755" alt="Screen Shot 2021-07-16 at 2 12 49 PM" src="https://user-images.githubusercontent.com/942359/125991977-c1bcef1f-4854-4c1a-b61b-f7d1c486a5a8.png">

**After:**
<img width="1755" alt="Screen Shot 2021-07-16 at 2 13 14 PM" src="https://user-images.githubusercontent.com/942359/125992037-fc5aa5b8-f7dd-4e0d-830d-ff30982e2f75.png">
<img width="1755" alt="Screen Shot 2021-07-16 at 2 13 37 PM" src="https://user-images.githubusercontent.com/942359/125992020-2c124f70-68d0-42ec-8b8c-7034c474bfdf.png">
<img width="1755" alt="Screen Shot 2021-07-16 at 2 13 26 PM" src="https://user-images.githubusercontent.com/942359/125992026-1a2bbebb-9c48-411f-96a4-d41b39281028.png">
<img width="1755" alt="Screen Shot 2021-07-16 at 2 13 20 PM" src="https://user-images.githubusercontent.com/942359/125992032-15b13109-e4e2-4776-ab54-de444e0c5283.png">

The biggest caveat here is the fact that users will most likely want to revert at some point. I know that Systems already manually handles some of these requests (when it's Automattician error) and it just simply doesn't scale. We'd either need to implement an ability to revert (more about that below), or be absolutely sure that Happiness doesn't support reverts on user request.

The script uses the less-destructive version of Headstart to add new content on top of your existing content. It also comes chock full of warnings before running it. I'd recommend gating it to a paid plan and adding an Activity Log entry for Headstart. As a bonus, you could maybe use the Headstart annotation script to save a quick annotation as a site option _before_ running the Headstart - then you could run that annotation to "revert" the Headstart if need-be after. We could try this approach with CLI commands to see how well it works before implementing the feature.

**Comment by donalirl** [comment]
>The biggest caveat here is the fact that users will most likely want to revert at some point.

FWIW, in almost four years and 34,000 user interactions, I have never encountered a customer who wanted to undo a headstart. The important thing is to make sure that the messaging in the UI is clear so that users:

* understand what headstart on demand is
* understand the consequences of taking the action
* help them decide if this option is right for them
* Understand that the action is irreversible

If that's all made clear, there shouldn't be any need to provide a revert option. At least, it doesn't seem like it should be a blocker for a V1 release.

I am happy to provide this messaging when the time comes as well.

**Comment by mpkelly** [comment]
As per this heads-up post, this feature is now available in Calypso. Closing this issue. 

p7DVsv-jA6-p2
